### DISCUSSION TITLE
Slices patterns, access multiple slices: data cannot be accessed

### DISCUSSION THREAD

**Discussion by norama** [pmndrs/zustand#2045] [Bug report] [answered]
## Summary

Using your multiple slices pattern, I tried to add a slice which acts on other slices, as in your docs:
https://docs.pmnd.rs/zustand/guides/slices-pattern#updating-multiple-stores
When I want to create a function acting on data in the slices then the default values are returned instead of the actual data.
Accessing the functions in the slices (as in your example) works fine, but how to access the data?
Also tried to add a `get` parameter when creating the slices according to
https://docs.pmnd.rs/zustand/recipes/recipes#read-from-state-in-actions
but this did not work for slices, they do not seem to get the `get` parameter.

## Link to reproduction

https://codesandbox.io/s/silly-tamas-2rp6gx?file=/src/store/index.ts

Reset works, but Calc sum always displays 0.

## Check List

Please do not ask questions in issues.

- [ x] I've already opened a [discussion](https://github.com/pmndrs/zustand/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ x] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

Please check this if you're filing an issue regarding TypeScript.

- [ x] I've read the [typescript guide](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md), in particular that `create` is to be used as `create<T>()(...)` and not `create<T>(...)`.

**Comment by dai-shi** [comment]
Can anyone help here?

**Reply by dbritto-dev** [reply]
@dai-shi I can help

**Comment by dbritto-dev** [comment, marked as answer]
@norama you are trying to create actions base on create slice functions which is wrong, that's why you always get the default values. You should always use `get()` to access to any function or prop/field.

Here you go

```ts
const createResetAllSlice = (set, get) => ({
  resetAll: () => {
    get().resetApples();
    get().resetPears();
  },
  sum: () => get().apples + get().pears
});
```

**Reply by norama** [reply]
@dbritto-dev Thanks a lot for the solution and the quick reaction!

It was also strange for me that we create new slices in order to access fields
(I also wonder why this works for functions at all), but this is what your docs says:
https://docs.pmnd.rs/zustand/guides/slices-pattern#updating-multiple-stores
So please correct the docs!
Also please include **typescript docs** for this because it is far from trivial as this slice has access to the whole store.

Now I realize that you have it elsewhere in the docs in a different way:
https://docs.pmnd.rs/zustand/guides/slices-pattern#slicing-the-store-into-smaller-stores
```javascript
export const createBearSlice = (set) => ({
  bears: 0,
  addBear: () => set((state) => ({ bears: state.bears + 1 })),
  eatFish: () => set((state) => ({ fishes: state.fishes - 1 })),
})
```
Here this slice directly refers to `state.fishes` which is in another slice.
I thought it is a bug in the docs but the `state` will be merged and the slice refers to the merged state.
Anyway, it would be nice if you could describe this in more detail in the docs.

**Reply by dbritto-dev** [reply]
@norama you could access state using `get()` or inside of `set()` callback fn

**Reply by dbritto-dev** [reply]
@norama about multiple stores you are right, the docs could misguide you I'll work on that but apart from that since we are talking about slices (same store/state) you can access to whatever field from the store/state. Just keep in mind talking about slices doesn't mean different stores just different fields in the same stores kind of namespace but that's it. So is not like you couldn't access other slice from a different slice.

**Reply by dbritto-dev** [reply]
@norama here you go -> https://github.com/pmndrs/zustand/pull/2046

**Reply by norama** [reply]
Thank you for the quick docs update, just some typescript example please.
